model/parsley/parsley.go

```diff
@@ -0,0 +1,43 @@
+package parsley
+
+import (
+	"regexp"
+
+	"github.com/mongodb/grip"
+)
+
+// ParsleyFilter represents a filter for the Parsley log viewer. Parsley filters can be defined at
+// the project-level and at the user-level.
+type ParsleyFilter struct {
+	Expression    string `bson:"expression" json:"expression"`
+	CaseSensitive bool   `bson:"case_sensitive" json:"case_sensitive"`
+	ExactMatch    bool   `bson:"exact_match" json:"exact_match"`
+}
+
+func (p ParsleyFilter) validate() error {
+	catcher := grip.NewSimpleCatcher()
+	catcher.NewWhen(p.Expression == "", "filter expression must be non-empty")
+
+	if _, regexErr := regexp.Compile(p.Expression); regexErr != nil {
+		catcher.Wrapf(regexErr, "filter expression '%s' is invalid regexp", p.Expression)
+	}
+
+	return catcher.Resolve()
+}
+
+// ValidateParsleyFilters checks that there are no duplicate expressions among the Parsley filters. It also validates
```

Comment:
Would it not be okay to have multiple filters with the same expression that have different sensitivity / match? Should we check that those aren't the same as well?

Reply:
Reading back to my tech design, I'm not sure why I introduced such a strict constraint. Possibly because it was discussed when we reviewed [EVG-18730](https://jira.mongodb.org/browse/EVG-18730) in design sync -- I agree that it seems not necessary though so I removed it